source/subs_functions/subs_functions.bas

```diff
@@ -1063,7 +1063,7 @@ id.args = 2
 id.arg = MKL$(STRINGTYPE - ISPOINTER) + MKL$(LONGTYPE - ISPOINTER)
 id.specialformat = "?[,?]"
 id.ret = LONGTYPE - ISPOINTER
-id.hr_syntax = "_LOADIMAGE(fileName$[, mode])"
+id.hr_syntax = "_LOADIMAGE(fileName$[, mode& | mode$])"

```

Comment:
Not to throw a huge wrench in things, but I had thought you'd just add a third argument to go along with bpp, rather than turn the second one into a string. TBH I don't know which is better, but having an optional third argument would probably be a fair amount simpler from an implementation standpoint 🤷

Reply:
Ya, we need both the second and the third argument to be optional.  That is what I originally intended to do. And honestly, that would have been far easier and simpler had it worked the way I expected it to.

Just to give a little background...
While I was working on the dialog box stuff, I figured that you cannot have more than one optional argument for functions. The `passed` argument works like a `bool` in the case of functions. It works like a bit field for Subs. Also, if you look at [subs_functions.bas](https://github.com/QB64-Phoenix-Edition/QB64pe/blob/main/source/subs_functions/subs_functions.bas), we do not have a single function that takes more than one optional argument. The only exception to that is `_SCREENIMAGE` but that makes all argument optional in one go and `func__screenimage` handles `passed` as a `bool`.

Let me give it a go again on the different branch. Probably, I'll be able to explain it better then.

Reply:
Well that is unfortunate, I did not realize that :-/ I wonder how hard it would be to fix that, it's probably not pretty though...